Assets/__Scripts/Platforms/Gaga/GagaDiskManager.cs

```diff
@@ -0,0 +1,164 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Beatmap.Base;
+using UnityEngine;
+
+public class GagaDiskManager : MonoBehaviour
+{
+    private readonly int minEventValue = 0;
+    private readonly int maxEventValue = 8;
+    private readonly int[] heightEventTypes = new[] { 18, 16, 12, 13, 17, 19 };
+    
+    public List<GagaDisk> Disks = new List<GagaDisk>();
+    private EventGridContainer eventGridContainer;
+    private AudioTimeSyncController atsc;
+    
+    private Dictionary<int, List<BaseEvent>> cachedHeightEvents = new Dictionary<int, List<BaseEvent>>();
+
+    public void Start()
+    {
+        atsc = FindObjectOfType<AudioTimeSyncController>();
+        eventGridContainer = FindObjectOfType<EventGridContainer>();
+        foreach (var disk in Disks)
+        {
+            // Start at Y 20 (default).
+            disk.Init(20); 
+            
+            // Init cache for each height event type.
+            UpdateEventCache(disk.HeightEventType);
+        }
+        
+        eventGridContainer.ObjectSpawnedEvent += UpdateEventCache;
+        eventGridContainer.ObjectDeletedEvent += UpdateEventCache;
+    }
+
+    public void OnDestroy()
+    {
+        eventGridContainer.ObjectSpawnedEvent -= UpdateEventCache;
+        eventGridContainer.ObjectDeletedEvent -= UpdateEventCache;
+    }
+
+    private void LateUpdate()
+    {
+        foreach (var disk in Disks)
+        {
+            disk.LateUpdateDisk(atsc.CurrentJsonTime);
+        }
+    }
+
+    public void HandlePositionEvent(BaseEvent evt)
+    {
+        var value = evt.Value;
```

Comment:
Unused variable.

Reply:
Yes. There is no point adding code that does nothing. (Issue is already resolved.)